lib/battle/BattleHex.h

```diff
@@ -87,6 +87,10 @@ struct DLL_LINKAGE BattleHex //TODO: decide if this should be changed to class f
 	/// returns all valid neighbouring tiles
 	std::vector<BattleHex> neighbouringTiles() const;
 
+	/// returns all valid (not first and last columns) neighbouring tiles, with their relative direction
+	std::vector<std::pair<BattleHex::EDir, BattleHex>> neighbouringTilesWithDirection() const;

```

Comment:
to draw the limit on the edge of the hexes, I need to know the edge (direction) where to draw. allNeighbouringTiles does not give me any direction information.

Reply:
Was looking at wrong function: neighbouringTiles :(
Ok, looks almost like what I need. I need only tiles that are valid... could I pass a parameter to this function?
```
std::vector<BattleHex> BattleHex::allNeighbouringTiles() const
{
	std::vector<BattleHex> ret;
	ret.resize(6);

	for(auto dir : hexagonalDirections())
		ret[dir] = cloneInDirection(dir, false);

	return ret;
}
```

Reply:
Don't think that parameter is needed, you can:
1) Add check inside this method. This method is used once and caller is only interested in valid hexes.
2) Add check inside your method.

Reply:
Added check in my method. Cannot add check inside because crash.
For reference: There are 2 distinct methods available: `allNeighbouringTiles `and `neighbouringTiles `this one returns only valid ones (but since is not all, direction cannot be inferred).